### ISSUE TITLE
Option to disable request batching

### ISSUE THREAD

**Issue by nikolai-mb** [gqty-dev/gqty#645]
Hi, we have been trying out this package, but quickly ran into an issue regarding the request scheduler / batcher which groups multiple requests into the same HTTP request.

If one of the batched queries takes significantly longer than the others, the other requests will also be delayed in their response.
Most of our queries are not related, and we would like these to be sent separately so that loading states can be handled on an individual basis.

We have tried tuning the client option "catchSelectionsTimeMS" but this does not seem to help in our case, since the queries we want to be sent as seperate HTTP requests are dispatched within the same render cycle, and are grouped regardless.

I initially thought that if we used multiple useQuery hook instances, that these would batch requests individually, but this does not seem to be the case, and it seems like batching is done globally.

Is there any way to disable the batch scheduler or to control the behavior on a per-hook instance basis?

We would really like to use this package, but the batching functionality is currently preventing us from doing so.
Any help is much appreciated!

**Comment by PabloSzx** [comment]
For this use case, and many others, there is `useTransactionQuery` and the core client `resolved`, these isolate the queries into it's own fetchs

**Comment by nikolai-mb** [comment]
We have tried out the useTransactionQuery hook in a few different places, but we have not found it usable.
As far as we have been able to tell, the fields you intend to use, must be specified up-front, and it's not detected by the proxy solution as seen in useQuery.

```
const { data, error, isLoading } = useTransactionQuery((query, args) => query.entity?.items?.map(i => ({ guid: i.guid, title: i.title })))

return (
  <div>
    {data?.map(i => (
      <div key={i.guid ?? ''}>
        {i.title}
      </div>
    ))
  </div>
)
```
^ Guid and Title props must be explicitly mapped in this example, or else we receive the "No data was requested" warning, as opposed to the useQuery hook where it's automatically detected by the proxy. This introduces a lot of boilerplate and is not a approach we want to go with. The mapping code gets exponentially more complex when we need to query fields on sub-collections. Is it not possible to use the transactionQuery hook without providing a lambda with the fields explicitly mapped?

It also seems like we need to use the onCompleted event to force a re-render after a variable has been changed, or else new data will be queried successfully, but the component will not re-render and display it. isLoading also seems to only work on first render (may be by-design).

The only thing we really need, is the useQuery hook functionality, without the request batching. The amount of extra code required in the other hooks sadly render them useless for us, and the behavior seems quite unpredictable.

**Comment by nikolai-mb** [comment]
I see, thank you for your insight on this. We have decided to go with a more traditional way of running graphql queries since we were unable to achieve what we desired in a stable way using the useTransactionQuery. We'll keep an eye on this library moving forward in case our situation should change.

I do find the library very impressive (the useQuery proxy aspect anyhow), so hopefully our paths will cross again.
